components/zcash_note_encryption/src/lib.rs

```diff
@@ -627,19 +631,24 @@ pub fn try_compact_note_decryption<D: Domain, Output: ShieldedOutput<D, COMPACT_
     try_compact_note_decryption_inner(domain, ivk, &ephemeral_key, output, &key)
 }
 
-fn try_compact_note_decryption_inner<D: Domain, Output: ShieldedOutput<D, COMPACT_NOTE_SIZE>>(
+/// This function is only meant to be used with compact notes, not full notes.
+/// If the note is a full note, then this function returns `None`.
+///
+/// For full notes, use 'try_note_decryption` and `try_note_decryption_inner`.
+fn try_compact_note_decryption_inner<D: Domain, Output: ShieldedOutput<D>>(
     domain: &D,
     ivk: &D::IncomingViewingKey,
     ephemeral_key: &EphemeralKeyBytes,
     output: &Output,
     key: &D::SymmetricKey,
 ) -> Option<(D::Note, D::Recipient)> {
     // Start from block 1 to skip over Poly1305 keying output
-    let mut plaintext = [0; COMPACT_NOTE_SIZE];
-    plaintext.copy_from_slice(output.enc_ciphertext());
+    let mut plaintext: D::CompactNotePlaintextBytes =
+        output.enc_ciphertext_compact().as_ref().into();
```

Comment:
At this stage `plaintext` is still en encrypted bytestring, even though it is of type `D::CompactNotePlaintextBytes`

Reply:
I'll stick with `plaintext` no minimizes the changes because `to_plaintext` creates the same issue but from the other side. (both comments)